### PR TITLE
Remove overrides

### LINKED ISSUE #109: Deprecate Happi
## Expected Behavior

There should be no explicit dependency on `happi` inside `pcdsdevices` . Happi loads devices with no explicit dependencies.

### LINKED ISSUE #114: Deprecate all enforced interfaces
## Expected Behavior

This is not Java. We will duck-type

### PR DESCRIPTION
**Pull request by teddyrendahl**
## Description

We should no longer subclass Ophyd Devices and Components. This PR fully switches everything away from `pcdsdevice` specific reimplementations. I am confident this will not break our live devices. There are a few cases where we will have to be more careful about when we instantiate PVs e.t.c, but it shouldn't be an issue if we follow the conventions laid out in #113 

## Motivation and Context

The overrides were all things that have now been pushed back to ophyd or covered up for a mistake elsewhere in the library. This reduces the level of complication in the code (All the tests pass and I deleted 1371 lines of code)

Closes #109 
Closes #114 

## How Has This Been Tested?

Almost all tests remained the same. That pesky PIM test fails again though. By default the `sim` tests aren't run, but you can use the `--sim` argument with `run_tests.py`

```
    @property
    def _readback(self):
        if not self.use_string:
            try:
>               return self._get_readback() + self._noise_func() * self.noise
E               TypeError: unsupported operand type(s) for +: 'NoneType' and 'float'
```

### CODE REVIEW

**Comment by ZLLentz** [comment]
Also, I suggest you remove `test_happi.py` to reflect the fact that this module won't be taking care of that any more.

**Comment by teddyrendahl** [comment]
Alright next wave:
* Remove `live_tests`. I like this concept but it belongs at hutch-level integration test
* Clean out all happi references
* Update `requirements.txt` and the conda-recipe

I realize I didn't mention this above, there is a strong possibility this breaks the `pswalker` tests. I think the `PIM` might have issues.

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -18,7 +18,6 @@ requirements:
   run:
     - python {{PY_VER}}*,>=3
     - ophyd >=1.0.0,<1.1
```
Comment (ZLLentz) [inline conda-recipe/meta.yaml]: Perhaps this is a good opportunity to unlock this requirement to be `>=1.0.0,<2` until we actually see a problem with a `1.x` release

File: tests_live/test_structure.py [hunk 2]
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-def test_explosions():
-    pass
```
Comment (ZLLentz) [inline tests_live/test_structure.py]: I hope you found the existence of this file as funny as I did

Comment (teddyrendahl) [reply tests_live/test_structure.py]: 😆

**Pull request merged**